packages/core/src/state-management/state-manager.ts

```diff
@@ -15,14 +15,22 @@ import {
 import { RunningState } from './running-state';
 import CID from 'cids';
 import { catchError, concatMap, takeUntil } from 'rxjs/operators';
-import { empty, Subscription, timer } from 'rxjs';
+import { Subscription, timer } from 'rxjs';
 import { SnapshotState } from './snapshot-state';
 import { CommitID, StreamID } from '@ceramicnetwork/streamid';
 
 const APPLY_ANCHOR_COMMIT_ATTEMPTS = 3
 
 export class StateManager {
 
+  /**
+   * Keeps track of every pinned StreamID that has had its state 'synced' (i.e. a query was sent to
+   * pubsub requesting the current tip for that stream) since the start of this process. This set
+   * only grows over time, in line with how many pinned streams get queried.
+   * @private
+   */
+  private readonly syncedPinnedStreams: Set<string>
```

Comment:
Not important. Could be `private readonly syncedPinnedStreams: Set<string> = new Set()`, and no logic in constructor.